Optimizer/Minuit2/MinuitIF.hpp

```diff
@@ -30,36 +22,29 @@ namespace Minuit2 {
 ///
 class MinuitIF : public Optimizer<MinuitResult> {
 public:
-  MinuitIF(bool UseHesse_ = true, bool UseMinos_ = false);
-
-  void useMinos(bool x);
-  void useHesse(bool x);
-
   MinuitResult optimize(ComPWA::Estimator::Estimator<double> &Estimator,
                         ComPWA::FitParameterList InitialParameters);
 
-private:
-  static FitParameterList
-  getFinalParameters(const ROOT::Minuit2::MnUserParameterState &minState,
-                     FitParameterList InitialParameters);
-  static std::vector<std::vector<double>>
-  getCovarianceMatrix(const ROOT::Minuit2::MnUserParameterState &minState);
-
-  bool UseHesse;
-  bool UseMinos;
+  bool UseHesse = 1;
+  bool UseMinos = 0;
+  
+  /// Minuit strategy (low, medium, high)
+  /// See https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.html#m-strategy
+  /// Settings are overwritten if specific parameters (e.g. GradientNCycles) are set
+  std::string MinuitStrategy = "medium";
+
+  // Minuit configuration variables
+  unsigned int GradientNCycles = 0;
+  double GradientStepTolerance = 0;
+  double GradientTolerance = 0;
+  unsigned int HessianNCycles = 0;
+  unsigned int HessianGradientNCycles = 0;
+  double HessianStepTolerance = 0;
+  double HessianG2Tolerance = 0;
```

Comment:
Generally it would be a bit easier UI if these members would be initialized to some standard values ("medium") as given in the `Minuit2Strategy`, and that these members are actually used in the minimization. The user can then modifiy them individually, or with some helper function which would set these values according to the string specification. Also the `MinuitStrategy` member would be dropped.

Because in this interface the user might get confused about the priority of the members, so that these variables will overwrite previous settings. 

A pity that using the `Minuit2Strategy` as a member makes the ROOT::Minuit2 library a PUBLIC dependency again...